### PR TITLE
docs(css-easing-function): Rewrite the "Values" section

### LINKED ISSUE #20138: Improve the linear() easing function information
This is a follow-on task from https://github.com/mdn/content/pull/19896 to:
- explain the linear() syntax (input and output values, percentages)
- add examples for linear()

Spec: https://drafts.csswg.org/css-easing-2/#the-linear-easing-function

### PR DESCRIPTION
**Pull request by dipikabh**
### Description

- This PR updates the descriptions in the "Values" section for consistency in style and format and to better reflect the syntax of `<easing-funcion>`.

    `<easing-function> = linear | <linear-easing-function> | <cubic-bezier-easing-function> | <step-easing-function>`

- Additionally, values called `<point-list>`, `<number-of-steps>`, and `<direction>` have been removed because they were not referenced in the the [Formal syntax](https://developer.mozilla.org/en-US/docs/Web/CSS/easing-function#formal_syntax) section or the [spec](https://drafts.csswg.org/css-easing/#easing-functions).

- Added another example in the "Linear easing function" section under "Description".

### Motivation

To align the "Values" section to match the parameter names and the order in which they appear in the [Formal syntax](https://developer.mozilla.org/en-US/docs/Web/CSS/easing-function#formal_syntax) section (Formal syntax on this page is up-to-date with the [spec](https://drafts.csswg.org/css-easing/#easing-functions)).

### Related issues and pull requests

Fixes https://github.com/mdn/content/issues/20138

### CODE REVIEW

File: files/en-us/web/css/easing-function/index.md [hunk 1]
```diff
@@ -36,79 +36,81 @@ step-end
...
 
-  - : Indicates that the interpolation jumps immediately to its final state, where it stays until the end. This keyword represents the easing function `steps(1, jump-start)` or `steps(1, start)`.
+    ![Graphs of "Input progress" to "Output progress", of which "step-start" shows a hollow origin and a horizontal line extending from (0, 1) to (1, 1); and "step-end" shows a horizontal line extending from the origin to (1, 0) (hollow) and a point at (1, 1).](step.svg)
 
-- `step-end`
+    `steps()`: This function accepts a strictly positive {{cssxref("&lt;integer&gt;")}} and a second optional `<step-position>`.
 
-  - : Indicates that the interpolation stays in its initial state until the end, at which point it jumps directly to its final state. This keyword represents the easing function `steps(1, jump-end)` or `steps(1, end)`.
+    - `<integer>`: Represents the number of equidistant intervals or 'steps' in the animation. It must be a positive integer greater than `0` unless the second parameter is `jump-none`, in which case, it must be a positive integer greater than `1`.
```
Comment (estelle) [inline files/en-us/web/css/easing-function/index.md:86]:
```suggestion
    - `<integer>`: Represents the number of equidistant intervals or 'steps'. It must be a positive integer greater than `0` unless the second parameter is `jump-none`, in which case, it must be a positive integer greater than `1`.
```

File: files/en-us/web/css/easing-function/index.md [hunk 2]
```diff
@@ -36,79 +36,81 @@ step-end
...
+    - `<step-position>`: Indicates when the animation jumps from one step to the next. The possible keyword values include:
```
Comment (estelle) [inline files/en-us/web/css/easing-function/index.md:87]:
```suggestion
    - `<step-position>`: Indicates specifies the timing of the first jump, last jump, or both. The possible keyword values include:
```
What i wrote might not be perfect, but the step position is about the first and last jumps, not each steps timing, which is what the original seems to indicate.

File: files/en-us/web/css/easing-function/index.md [hunk 3]
```diff
@@ -36,79 +36,81 @@ step-end
...
 
-![Graphs of "Input progress" to "Output progress", of which "step-start" shows a hollow origin and a horizontal line extending from (0, 1) to (1, 1); and "step-end" shows a horizontal line extending from the origin to (1, 0) (hollow) and a point at (1, 1).](step.svg)
+      - `jump-start` denotes that the first step or jump happens when the interpolation begins.
```
Comment (estelle) [inline files/en-us/web/css/easing-function/index.md:89]:
```suggestion
      - `jump-start` denotes that the first step or jump happens when the interpolation begins, essentially jumping past the `0` point.
```

File: files/en-us/web/css/easing-function/index.md [hunk 4]
```diff
@@ -36,79 +36,81 @@ step-end
...
+      - `jump-end` denotes that the last step or jump happens when the interpolation ends. This is the default value when no `<step-position>` is specified.
```
Comment (estelle) [inline files/en-us/web/css/easing-function/index.md:90]:
```suggestion
      - `jump-end` denotes that the last step or jump happens when the interpolation ends, staying on `0%` but no duration of time at `100%`. This is the default value when no `<step-position>` is specified.
```

File: files/en-us/web/css/easing-function/index.md [hunk 5]
```diff
@@ -36,79 +36,81 @@ step-end
...
+      - `jump-none` denotes that no jump happens at the beginning or the end, effectively removing a step during the interpolation iteration. Instead, the animation holds steady at both the 0% mark and the 100% mark. The duration of these holds is determined by dividing the total animation duration by the number of steps (1/n).
+      - `jump-both` denotes that jumps occur at both the 0% and 100% marks, effectively adding a step during the interpolation iteration.
```
Comment (estelle) [inline files/en-us/web/css/easing-function/index.md:92]:
```suggestion
      - `jump-both` denotes that jumps occur at both ends, spending to time at the 0% and 100% marks, effectively adding a step during the interpolation iteration.
```

**Review by estelle — CHANGES_REQUESTED** [review]
The easing function is used for both animations and transitions, so we need to watch our usage of the animation term

File: files/en-us/web/css/easing-function/index.md (hunk 2 above)
Comment (dipikabh) [reply files/en-us/web/css/easing-function/index.md:87]: Thanks for pointing out. Updating it to:

> Specifies the timing of the jump either at the start, the end, or both.

File: files/en-us/web/css/easing-function/index.md (hunk 3 above)
Comment (dipikabh) [reply files/en-us/web/css/easing-function/index.md:89]: Thanks for the hint. Updating this to:

> - `jump-start` denotes that the first step occurs right at the start, essentially beginning at the `0` point.

File: files/en-us/web/css/easing-function/index.md (hunk 4 above)
Comment (dipikabh) [reply files/en-us/web/css/easing-function/index.md:90]: To keep the structure similar to that of `jump-start`, how about rewording `jump-end` to:

> - `jump-end` denotes that the last step occurs right at the end, essentially jumping at the `1` point at the end of the duration. This is the default value when no `<step-position>` is specified.

File: files/en-us/web/css/easing-function/index.md (hunk 5 above)
Comment (dipikabh) [reply files/en-us/web/css/easing-function/index.md:92]: Updating this one to:

>  - `jump-both` denotes that jumps occur at both the beginning and the end, essentially occurring at both the `0` and `1` points. Effectively, a step is added at both the ends. No time is spent at the `0%` and `100%` marks.

File: files/en-us/web/css/easing-function/index.md [hunk 6]
```diff
@@ -21,7 +21,7 @@ linear
 
 /* cubic-bezier function and keywords */
 /* cubic-bezier(<x1>, <y1>, <x2>, <y2>) */
-cubic-bezier(0.42, 0.0, 1.0, 1.0)
+cubic-bezier(0.42, 0, 1, 1)
```
[the remark this answers is not in the document]
Comment (dipikabh) [reply files/en-us/web/css/easing-function/index.md:24]: Thanks, I've updated the syntax example to `cubic-bezier(0.25, 0.1, 0.25, 1)`.

> might as well add ease-in-out to the list on line 28, that way they're all listed.

`ease-in-out` exists in the list on line 28, unless I am missing something

File: files/en-us/web/css/easing-function/index.md [hunk 7]
```diff
@@ -36,79 +36,81 @@ step-end
...
 ### Values
 
-- `<point-list>`
-
-  - : List of linear stops.
-
 - `linear`
 
-  - : Indicates that the interpolation goes linearly. This keyword represents the easing function `linear(0, 1)`.
+  - : Specifies a constant rate of interpolation, ensuring there is no change in the rate of progress of the animation throughout the duration. This also means that there is no acceleration or deceleration at any point during the animation's duration. It's important to not confuse this keyword value with the [linear() easing function](#linear_easing_function), although they are related: the `linear` keyword produces a `linear()` easing function with two points, `linear(0, 1)`.
```
[the remark this answers is not in the document]
Comment (dipikabh) [reply files/en-us/web/css/easing-function/index.md:41]: Updated as:

> Specifies a constant rate of interpolation, with no change in the rate of progress throughout the duration (that is, no acceleration or deceleration). This keyword value is equivalent to both [`cubic-bezier(0, 0, 1, 1)`](#cubic_bézier_easing_function) and [`linear(0, 1)`](#linear_easing_function).

File: files/en-us/web/css/easing-function/index.md [hunk 8]
```diff
@@ -36,79 +36,81 @@ step-end
...
+    > **Note:** The `linear` keyword always serializes as `linear`, whereas its function equivalent `linear(0, 1)` serializes as `linear(0 0%, 1 100%)`.
```
[the remark this answers is not in the document]
Comment (dipikabh) [reply files/en-us/web/css/easing-function/index.md:42]: I was basing these updates on the `linear` keyword is explained in the spec: https://drafts.csswg.org/css-easing/#valdef-easing-function-linear.

To keep this keyword description simple as suggested (I agree), I've moved the Note to the [Linear easing function](https://pr29585.content.dev.mdn.mozit.cloud/en-US/docs/Web/CSS/easing-function#linear_easing_function) section. Also simplified "serializes" to "interpreted as":

> **Note:** The [`linear`](#linear) keyword is equivalent to the easing function `linear(0, 1)`. While the `linear` keyword is always interpreted as-is, the function `linear(0, 1)` gets interpreted as `linear(0 0%, 1 100%)`.

File: files/en-us/web/css/easing-function/index.md [hunk 9]
```diff
@@ -36,79 +36,81 @@ step-end
...
 
 ![Graph of "Input progress" to "Output progress" showing a line extending from the origin to (1, 1).](linear.svg)
 
-- `<x1>`, `<y1>`, `<x2>`, `<y2>`
+- `<linear-easing-function>`
+
+  - : Specifies a `linear()` function using multiple points, called linear stops, allowing you to control the movement of an animation at specific points along the path.
+
+    The `linear()` function specifies a list of points (`<linear-stop-list>`), separated by commas. Each point or linear stop in the list (`<linear-stop>`) is specified as a {{cssxref("&lt;number&gt;")}} and optional one or two {{cssxref("&lt;percentage&gt;")}} values (`<linear-stop-length>`).
```
[the remark this answers is not in the document]
Comment (dipikabh) [reply files/en-us/web/css/easing-function/index.md:50]: Thanks for helping with simplifying Updated this to:

> Specifies a `linear()` function with one or more comma-separated _linear stops_, each containing up to two optional _stop lengths_, to control the progress of an animation or transition.

>  The `linear()` function specifies a `<linear-stop-list>`, a comma-separated list of points along the animation or transition progress. Each point or `<linear-stop>` in the list is specified as a {{cssxref("&lt;number&gt;")}} between `0` and `1`. By default, each stop in the `<linear-stop-list>` is equidistant. To have more control on the progress of the animation or transition, each point can include up to two optional {{cssxref("&lt;percentage&gt;")}} `<linear-stop-length>` values.

File: files/en-us/web/css/easing-function/index.md [hunk 10]
```diff
@@ -36,79 +36,81 @@ step-end
...
 
-![Graphs of "Input progress" to "Output progress", of which "ease" shows a curved line quickly rising from the origin to (1, 1); "ease-in" shows a shallow curved line from the origin that straightens out as it approaches (1, 1); "ease-out" shows a straight diagonal line that slightly curves as it gets close to (1, 1); and "ease-in-out" shows a symmetrical, "S"-shaped line curving from the origin to (1, 1).](ease.svg)
+    `cubic-bezier()`: This function accepts four {{cssxref("&lt;number&gt;")}} values.
 
-- `<number-of-steps>`
+    - `<number>`: Specifies the abscissas and ordinates of the [P1 and P2 points](#cubic_bézier_easing_function) of the curve (as `<x1>`, `<y1>`, `<x2>`, and `<y2>`). `<x1>` and `<x2>` must be in the range `[0, 1]`, otherwise the value is invalid.
```
[the remark this answers is not in the document]
Comment (dipikabh) [reply files/en-us/web/css/easing-function/index.md:72]: 🙂 How does this look:

> - {{cssxref("&lt;number&gt;")}}: Specifies the location of [P1 and P2 points](#cubic_bézier_easing_function) on the curve. `<x1>` and `<y1>` are the coordinates for point P1, and `<x2>` and `<y2>` are the coordinates for point P2. `<x1>` and `<x2>` values must be between `0` and `1`, otherwise the function will not work as expected.

File: files/en-us/web/css/easing-function/index.md [hunk 11]
```diff
@@ -36,79 +36,81 @@ step-end
...
+    - `<number>`: Represents a point in time. The value should be between `0` and `1`.
 
-- `ease`
+    - `<percentage>`: Represents the progress of the animation at the specified point in time. If two percentage values are specified, they represent a range for that specific point – the first percentage indicates the starting point and the second percentage indicates the ending point of the progress for that segment in the animation. If no `<percentage>` value is specified, the points are spread evenly along the timeline.
 
-  - : Indicates that the interpolation starts slowly, accelerates sharply, and then slows gradually towards the end. This keyword represents the easing function `cubic-bezier(0.25, 0.1, 0.25, 1.0)`. It is similar to [`ease-in-out`](#ease-in-out), though it accelerates more sharply at the beginning.
+- `<cubic-bezier-easing-function>`
 
-- `ease-in`
+  - : Specifies a `cubic-bezier()` function using two control points, which help shape the curve of the animation's progress. This parameter accepts one of the following four keyword values that map to predefined `cubic-bezier()` functions or a custom `cubic-bezier()` function to specify the control points:
```
[the remark this answers is not in the document]
Comment (dipikabh) [reply files/en-us/web/css/easing-function/index.md:58]: Thanks for the hints! I've rephrased this as:

> Specifies a `cubic-bezier()` function using two control points, each with two values, which help shape the curve of the animation or transition progress. The control points can be specified using a custom `cubic-bezier()` function, or one of the four predefined keyword values for the `cubic-bezier()` function can be used. The predefined keywords include:

**Comment by dipikabh** [comment]
Thanks very much @estelle for your valuable feedback.
I have addressed all your comments. Let me know what you think.

File: files/en-us/web/css/easing-function/index.md [hunk 12]
```diff
@@ -36,81 +36,83 @@ step-end
...
+    - `<percentage>`: Indicates the position of a linear stop along the duration. It can take up to two values. If one value is specified, it defines the start of the associated linear stop. If two percentage values are specified, they define the length of the stop: the first percentage indicates the starting point and the second percentage indicates the ending point for that segment in the animation or transition. If no `<percentage>` value is specified, which is the default, the stops are spread evenly along the timeline.
 
-- `ease-in`
+- `<cubic-bezier-easing-function>`
 
-  - : Indicates that the interpolation starts slowly, then progressively speeds up until the end, at which point it stops abruptly. This keyword represents the easing function `cubic-bezier(0.42, 0.0, 1.0, 1.0)`.
+  - : Specifies a `cubic-bezier()` function using two control points, each with two values, which help shape the curve of the animation or transition progress. The control points can be specified using a custom `cubic-bezier()` function, or one of the four predefined keyword values for the `cubic-bezier()` function can be used. The predefined keywords include:
```
Comment (estelle) [inline files/en-us/web/css/easing-function/index.md:57]:
```suggestion
  - : Specifies a Bézier curve that shapes the progression of the animation or transition.  In CSS, [Bézier curves] are mathematically described curves defined by four control points. The cubic bezier easing function is defined either via either one of the keyword values mapped to common bezier curve parameter values or, to create a custom curve, via a four-parameter `cubic-bezier()` function call. The four predefined keywords include:
```

**Review by estelle — APPROVED** [review]
Should we add a `steps(10, jump-both)` near line 32 so that there is a jump example?

Should we create `linear()` and `cubic-bezier()` function pages? The CSS functions page leads here, so at least they are listed in functions, but (almost) all the other functions on that page have their own page. I think doing that would afford us more room to explain them with examples while making the sections on this page more compact and understandable / less complicated.

Otherwise, looks good. just made a suggestion for bezier. if you don't agree with it, that's fine, but please do include the link to the glossary page. Approving to not hold this up, and get the info out there, but i think we should create the function pages sooner rather than later.

File: files/en-us/web/css/easing-function/index.md (hunk 12 above)
Comment (dipikabh) [reply files/en-us/web/css/easing-function/index.md:57]: Thanks! The suggestion adds more clarity. I've updated the text as:

> - : Specifies a [Bézier curve](/en-US/docs/Glossary/Bezier_curve) to shape the progress of an animation or a transition. In CSS, Bézier curves are defined by four control points that mathematically describe the curve: a starting point, an ending point, and two control points. The cubic Bézier easing function can be defined in one of these two ways: by creating a custom curve with a four-parameter `cubic-bezier()` function call or by using one of the predefined keyword values, which map to the commonly used Bézier curve parameters. The predefined keyword values include:

Also added link to glossary here. The term is linked to glossary in "Description" > "Cubic Bézier easing function" section as well.

**Comment by dipikabh** [comment]
> Should we add a steps(10, jump-both) near line 32 so that there is a jump example?

Good idea. I've added it.

> Should we create linear() and cubic-bezier() function pages? The CSS functions page leads here, so at least they are listed in functions, but (almost) all the other functions on that page have their own page. I think doing that would afford us more room to explain them with examples while making the sections on this page more compact and understandable / less complicated.

Makes sense! I'll open an issue to track this work.

> Otherwise, looks good. just made a suggestion for bezier. if you don't agree with it, that's fine, but please do include the link to the glossary page. Approving to not hold this up, and get the info out there, but i think we should create the function pages sooner rather than later.

Thanks for the suggestion! I've accepted it with a little tweak. Thanks for the review and approving!
